src/OpenTelemetry.Instrumentation.Dependencies/GrpcClientInstrumentation.cs

```diff
@@ -0,0 +1,45 @@
+﻿// <copyright file="GrpcClientInstrumentation.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+using System;
+using OpenTelemetry.Instrumentation.Dependencies.Implementation;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Instrumentation.Dependencies
+{
+    /// <summary>
+    /// GrpcClient instrumentation.
+    /// </summary>
+    public class GrpcClientInstrumentation : IDisposable
+    {
+        private readonly DiagnosticSourceSubscriber diagnosticSourceSubscriber;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GrpcClientInstrumentation"/> class.
+        /// </summary>
+        /// <param name="activitySource">ActivitySource adapter instance.</param>
+        public GrpcClientInstrumentation(ActivitySourceAdapter activitySource)
```

Comment:
Check activitySource is not null? e.g.

```
if (activitySource == null)
{
    throw new ArgumentNullException(nameof(activitySource));
}
```

Reply:
Seems like a good idea, so I made the change. Though it raises the question of whether the factory that instantiates the instrumentation should be defensive. Anyone else have thoughts here?

Here's where it's instantiated:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/src/OpenTelemetry/Trace/Configuration/OpenTelemetrySdk.cs#L85-L93

Reply:
Agree that the factory should do more checks.
Also wondering why make GrpcClientInstrumentation (and similar classes for http etc) public? Users enable/configure instrumentation using OpenTelemetryBuilder extension methods only.

Reply:
I made GrpcClientInstrumentation internal. I could follow up in a small PR to make the other classes internal as well.